iOS.FrameMetrics/FrameMetricsReporter.cs

```diff
@@ -0,0 +1,69 @@
+using CoreAnimation;
+using CoreFoundation;
+using Foundation;
+
+namespace iOS.FrameMetrics;
+
+/// <summary>
+/// Code here is loosely based on:
+/// https://thisiskyle.me/posts/measuring-ios-scroll-performance-is-tough-use-this-to-make-it-simple-and-automated.html
+/// </summary>
+public static class FrameMetricsReporter
+{
+    static CADisplayLink displayLink = null!;
+    static DispatchQueue queue = null!;
+    static double lastTimestamp;
+    static int currentFrameDropCount;
+
+    public static void Initialize()
+    {
+        displayLink = CADisplayLink.Create(OnFrame);
+        displayLink.Paused = true;
+        displayLink.AddToRunLoop(NSRunLoop.Main, NSRunLoopMode.Common);
+        queue = DispatchQueue.GetGlobalQueue(DispatchQueuePriority.High);
+
+        // Start the display link
+        displayLink.Paused = false;
+    }
+
+    static void OnFrame()
+    {
+        queue.DispatchAsync(() =>
```

Comment:
Do I actually need a `DispatchQueue` here? Seems like `CADisplayLink` would call `OnFrame()` appropriately?

Reply:
I think this is just to avoid doing stuff on the main thread (and thus interferring with the render loop).